.buildkite/test_cuda_mpi.jl

```diff
@@ -0,0 +1,89 @@
+#= Test basic CUDA and CUDA-aware MPI functionality
+
+Main testing functions:
+- `test_gpu_visbility`: Test access to at least one GPU and perform a minimal computation
+- `test_mpi`: Test a send/receive between arrays on MPI ranks. This is used to
+test transfers between CPU-allocated Arrays and GPU-allocated CuArrays.
+
+For performance, is important to verify that transfers between CuArrays on different GPUs 
+are peer-to-peer transfers, which is done by profiling the test with Nsight Systems.
+
+To run this test, you need 2 CUDA devices, CUDA-aware MPI, and Nsight Systems.
+1. mpirun -n 2 nsys profile --trace=nvtx,cuda,mpi julia --project bin/test_cuda_mpi.jl 
+2. nsys stats --report cuda_gpu_trace report1.nsys-rep
```

Comment:
Could you describe the expected output for a successful test?

Reply:
Here is the new comment in the test file:

#= Test basic CUDA and CUDA-aware MPI functionality

Main testing functions:

- `test_gpu_visbility`: Test access to at least one GPU and perform a minimal computation
- `test_mpi`: Test a send/receive between arrays on MPI ranks. This is used to
test transfers between CPU-allocated Arrays and GPU-allocated CuArrays.

For performance, is important to verify that transfers between CuArrays on different GPUs 
are peer-to-peer transfers, which is done by profiling the test with Nsight Systems.

To run this test, you need 2 CUDA devices, CUDA-aware MPI, and Nsight Systems.

First, run the test with two MPI ranks, profiling with Nsight:

    `mpirun -n 2 nsys profile --trace=nvtx,cuda,mpi julia --project bin/test_cuda_mpi.jl`

If this runs successfully, the test prints MPI and CUDA version info and the following output:

```
Testing GPU Visibility
Basic GPU computation successful
Testing MPI with Array
Transfer successful
Testing MPI with CuArray
Transfer successful
All tests completed!
```

Once the profile has been generated, analyze the data to determine if a 
peer-to-peer transfer occured between the GPUs:

`nsys stats --report cuda_gpu_trace report1.nsys-rep | grep grep -E "memcpy (Peer-to-Peer|PtoP)"`

You should see `[CUDA memcpy Peer-to-Peer]` if the peer-to-peer transfer was successful.
If you don't see this, you can view the full profile analysis using

`nsys stats --report cuda_gpu_trace report1.nsys-rep`
=#